### PR TITLE
terraform: 1.0.11 -> 1.1.0

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

https://github.com/hashicorp/terraform/releases/tag/v1.1.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
Guess `1.1.0` needs to be added as another version.

```diff
diff --git a/pkgs/applications/networking/cluster/terraform/default.nix b/pkgs/applications/networking/cluster/terraform/default.nix
index 036ee7d8243..a5f2e351cd2 100644
--- a/pkgs/applications/networking/cluster/terraform/default.nix
+++ b/pkgs/applications/networking/cluster/terraform/default.nix
@@ -202,6 +202,14 @@ rec {
     passthru = { inherit plugins; };
   };
 
+  terraform_1_1 = mkTerraform {
+    version = "1.1.0";
+    sha256 = "sha256-nnYMoQitqFbOjI8twDh9hWDb1qxMNNVy6wldxkyDKY0=";
+    vendorSha256 = "sha256-inPNvNUcil9X0VQ/pVgZdnnmn9UCfEz7qXiuKDj8RYM=";
+    patches = [ ./provider-path-0_15.patch ];
+    passthru = { inherit plugins; };
+  };
+
   # Tests that the plugins are being used. Terraform looks at the specific
   # file pattern and if the plugin is not found it will try to download it
   # from the Internet. With sandboxing enable this test will fail if that is
@@ -211,7 +219,7 @@ rec {
       mainTf = writeText "main.tf" ''
         resource "random_id" "test" {}
       '';
-      terraform = terraform_1_0.withPlugins (p: [ p.random ]);
+      terraform = terraform_1_1.withPlugins (p: [ p.random ]);
       test =
         runCommand "terraform-plugin-test" { buildInputs = [ terraform ]; } ''
           set -e
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index b91eac47b72..9618ff6952e 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -33140,10 +33140,11 @@ with pkgs;
     terraform_0_14
     terraform_0_15
     terraform_1_0
+    terraform_1_1
     terraform_plugins_test
     ;
 
-  terraform = terraform_1_0;
+  terraform = terraform_1_1;
   # deprecated
   terraform-full = terraform.full;
 
```

**Review by Chili-Man — APPROVED** [review]

**Review by snoord — APPROVED** [review]

**Comment by snoord** [comment]
Result of `nixpkgs-review pr 149720` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>infracost</li>
    <li>terraform (terraform_1_0)</li>
  </ul>
</details>

**Review by zowoq — APPROVED** [review]

**Comment by zimbatm** [comment]
Now that terraform has stabilized, we could have only `terraform_1`.

**Review by timstott — APPROVED** [review]

**Comment by marsam** [comment]
>Now that terraform has stabilized, we could have only terraform_1.

I agree. I've updated the pull request PTAL

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -930,6 +930,7 @@ mapAliases ({
   telnet = inetutils; # added 2018-05-15
   terminus = throw "terminus has been removed, it was unmaintained in nixpkgs"; # added 2021-08-21
   terraform_1_0_0 = terraform_1_0; # added 2021-06-15
+  terraform_1_0 = terraform_1; # added 2021-12-08
```
Comment (Ma27) [inline pkgs/top-level/aliases.nix:933]: This seems somewhat unintuitive for me now that terraform_1 is actually 1.1 rather than 1.0.

Comment (marsam) [reply pkgs/top-level/aliases.nix:933]: would you prefer `terraform_1_x`?

Comment (Ma27) [reply pkgs/top-level/aliases.nix:933]: I mean, the fact that using `terraform_1_0` now installs a Terraform 1.1 is a problem IMHO. `terraform_1` is perfectly fine though.

Comment (zimbatm) [reply pkgs/top-level/aliases.nix:933]: It's the same issue with `terraform_1_0_0`. Let's `throw "renamed to terraform_1"` instead for both.

Comment (marsam) [reply pkgs/top-level/aliases.nix]: Sorry for the delay, I've updated the aliases PTAL

**Comment by zimbatm** [comment]
thanks!

**Pull request merged**